Specification CI job failing

Specification CI job failing on main branch.

```
Run gem update --system
ERROR:  Error installing rubygems-update:
	There are no versions of rubygems-update (= 3.5.1) compatible with your Ruby & RubyGems
	rubygems-update requires Ruby version >= 3.0.0. The current ruby version is 2.7.8.225.
ERROR:  While executing gem ... (NoMethodError)
    undefined method `version' for nil:NilClass
Updating rubygems-update
Error: Process completed with exit code 1.
```

* https://github.com/lampepfl/dotty/actions/runs/7240389886/job/19723319886

#### Solution

Probably update ruby version to 3.0.0 or greater. Maybe need to update other versions as well.

--- a/.github/workflows/spec.yml
+++ b/.github/workflows/spec.yml
@@ -29,7 +29,8 @@ jobs:
           ruby-version: '2.7'
       - name: Install required gems
         run: |
-          gem update --system
+          gem install "rubygems-update:<3.5" --no-document
+          update_rubygems
           gem install sass-embedded -v 1.58.0
           gem install bundler:1.17.2 jekyll
           bundle install

